NitroxPatcher/Patches/Dynamic/Constructable_Construct_Patch.cs

```diff
@@ -1,62 +1,185 @@
             {
-                return true;
+                return InstructionsToAdd;
             }
+            return null;
+        });
 
-            // If we are constructing a base piece then we'll want to store all of the BaseGhost information
-            // as it will not be available when the construction hits 100%
-            if (__instance is ConstructableBase constructableBase)
+    public static void ConstructionAmountModified(Constructable constructable)
+    {
+        // We only manage the amount change, not the deconstruction/construction action
+        if (!constructable.TryGetNitroxId(out NitroxId entityId))
+        {
+            // TODO: Maybe delete the ghost
+            Log.Error($"Couldn't find a NitroxEntity on {constructable.name}");
+            return;
+        }
+        float amount = NitroxModel.Helper.Mathf.Clamp01(constructable.constructedAmount);
+        /*
+         * Different cases:
+         * - Normal module (only Constructable), just let it go to 1.0f normally
+         * - ConstructableBase:
+         *   - if it's already in a base, simply update the current base
+         *   - else, create a new base
+         */
+        if (amount == 1f)
+        {
+            if (constructable is ConstructableBase constructableBase)
             {
-                BaseGhost baseGhost = constructableBase.gameObject.GetComponentInChildren<BaseGhost>();
-                if (baseGhost != null)
-                {
-                    lastTargetBase = baseGhost.TargetBase;
-                    lastTargetBaseOffset = baseGhost.TargetOffset;
-                }
-
-                lastFace = baseGhost switch
-                {
-                    BaseAddFaceGhost { anchoredFace: { } } baseAddFaceGhost => baseAddFaceGhost.anchoredFace.Value,
-                    BaseAddModuleGhost { anchoredFace: { } } baseAddModuleGhost => baseAddModuleGhost.anchoredFace.Value,
-                    _ => lastFace
-                };
+                CoroutineHost.StartCoroutine(BroadcastObjectBuilt(constructableBase, entityId));
+                return;
             }
-            else
+            IEnumerator postSpawner = BuildingPostSpawner.ApplyPostSpawner(constructable.gameObject, entityId);
+            // Can be null if no post spawner is set for the constructable's techtype
+            if (postSpawner != null)
             {
-                lastTargetBase = null;
-                lastTargetBaseOffset = default(Int3);
+                CoroutineHost.StartCoroutine(postSpawner);
             }
+        }
+        // update as a normal module
 
-            return true;
+        // TODO: A latest packet is always sent even after the base is fully built, we don't want that
+        // When #2021 is merged, just use ThrottledPacketSender.RemovePendingPackets
+        Resolve<ThrottledPacketSender>().SendThrottled(new ModifyConstructedAmount(entityId, amount),
+            (packet) => { return packet.GhostId; }, 0.1f);
```

Comment:
```suggestion
        // TODO: When #2021 is merged, just use ThrottledPacketSender.RemovePendingPackets
        Resolve<ThrottledPacketSender>().SendThrottled(new ModifyConstructedAmount(entityId, amount), packet => packet.GhostId, 0.1f);
```

Reply:
I took the function from the said PR and implemented it in.